pvanalytics/system.py

```diff
@@ -401,3 +403,177 @@ def infer_orientation_daily_peak(power_or_poa, sunny, tilts,
                 best_azimuth = azimuth
                 best_tilt = tilt
     return best_azimuth, best_tilt
+
+
+def _power_residuals_from_clearsky(system_params,
+                                   ghi, dhi, dni,
+                                   solar_zenith, solar_azimuth,
+                                   power_ac, temperature,
+                                   wind_speed,
+                                   temperature_coefficient,
+                                   temperature_model_parameters):
+    """Return the residuals between a system with parameters given in `params`
+    and the data in `power_ac`.
+
+    Parameters
+    ----------
+    system_params : array-like
+        array of four floats: tilt, azimuth, DC capacity, and inverter
+        DC input limit.
+    ghi : Series
+        Clear sky GHI
+    dhi : Series
+        Clear sky DHI
+    dni : Series
+        Clear sky DNI
+    solar_zenith : Series
+        Solar zenith at the same times as data in `power_ac`
+    solar_azimuth : Series
+        Solar azimuth at the same times as data in `power_ac`
+    power_ac : Series
```

Comment:
Purely editorial, better if `power_ac` precedes `solar_zenith` and `solar_azimuth`

Reply:
I think I'd like this to be either first or last, would you be okay with `power_ac`, `ghi`, ..., `solar_zenith`?

Reply:
Yes

Reply:
On second thought after the clear sky irradiance is fine. Not really important where it goes, since it is only every used as a kwarg by `scipy.optimize.least_squares`.